### PR TITLE
SDIT-838 Send Nomis course schedule ids for updates

### PR DESCRIPTION
**Pull request by mikehalmamoj**
See the [last commit](https://github.com/ministryofjustice/hmpps-prisoner-to-nomis-update/pull/311/commits/4d0e6d5cc89853a7ee5f77a2a426be101402c3d0) to avoid noise from an OpenAPI spec update and refactoring.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisIntTest.kt [hunk 1]
```diff
@@ -88,9 +89,11 @@ class ActivityToNomisIntTest : SqsIntegrationTestBase() {
           .withRequestBody(matchingJsonPath("minimumIncentiveLevelCode", equalTo("BAS")))
           .withRequestBody(matchingJsonPath("programCode", equalTo("LEISURE_SOCIAL")))
           .withRequestBody(matchingJsonPath("payPerSession", equalTo("F")))
+          .withRequestBody(matchingJsonPath("schedules[0].id", absent()))
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisIntTest.kt:92]: When creating schedules there is no id to send - doesn't exist yet

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisIntTest.kt [hunk 2]
```diff
@@ -340,10 +343,12 @@ class ActivityToNomisIntTest : SqsIntegrationTestBase() {
             .withRequestBody(matchingJsonPath("scheduleRules[1].endTime", equalTo("14:25")))
             .withRequestBody(matchingJsonPath("scheduleRules[1].tuesday", equalTo("true")))
             .withRequestBody(matchingJsonPath("scheduleRules[1].thursday", equalTo("false")))
+            .withRequestBody(matchingJsonPath("schedules[0].id", equalTo("$NOMIS_CRS_SCH_ID")))
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisIntTest.kt:346]: When updating a schedule send the Nomis course schedule id found form the mapping API

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/SchedulesIntTest.kt [hunk 3]
```diff
@@ -36,6 +36,7 @@ class SchedulesIntTest : SqsIntegrationTestBase() {
       await untilAsserted {
         NomisApiExtension.nomisApi.verify(
           putRequestedFor(urlEqualTo("/activities/$NOMIS_CRS_ACTY_ID/schedule"))
+            .withRequestBody(matchingJsonPath("id", equalTo("$NOMIS_CRS_SCH_ID")))
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/SchedulesIntTest.kt:39]: When updating a single course schedule send its id as found from the mapping API

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**